src/modules/Game.class.js

```diff
@@ -1,68 +1,277 @@
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+    this.initialState = initialState || this.generateNewBoard();
+    this.board = JSON.parse(JSON.stringify(this.initialState));
+    this.firstState = JSON.parse(JSON.stringify(this.board));
+    this.score = 0;
+    this.status = 'idle';
+    this.checkLeft = true;
+    this.checkRight = true;
+    this.checkUp = true;
+    this.checkDown = true;
+    this.checkStart = false;
+  }
+
+  // #region get states
+
+  getState() {
+    return this.board;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  // #endregion
+
+  // #region navigation
+
+  moveLeft() {
+    if (this.checkStart) {
+      const currentState = [...this.getState()];
+
+      this.board = this.board.map((row) => {
+        let newRow = row.filter((el) => el !== 0);
+
+        for (let i = 0; i < newRow.length; i++) {
+          if (newRow[i] === newRow[i + 1]) {
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow[i + 1] = 0;
+          }
+        }
+
+        newRow = newRow.filter((el) => el !== 0);
+
+        while (newRow.length < 4) {
+          newRow.push(0);
+        }
+
+        return newRow;
+      });
+
+      if (this.checkWin()) {
+        this.status = 'win';
+      }
+
+      if (this.differenceArrays(currentState, this.getState())) {
+        this.checkLeft = true;
+      } else {
+        this.checkLeft = false;
+      }
+
+      if (this.checkLeft) {
+        this.addRandomTile(this.board);
+      }
+
+      if (this.checkLose() && !this.checkWin()) {
+        this.status = 'lose';
+      }
+    }
+  }
+
+  moveRight() {
+    if (this.checkStart) {
+      const currentState = [...this.getState()];
+
+      this.board = this.board.map((row) => {
+        let newRow = row.filter((el) => el !== 0).reverse();
+
+        for (let i = 0; i < newRow.length; i++) {
+          if (newRow[i] === newRow[i + 1]) {
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow[i + 1] = 0;
+          }
+        }
+
+        newRow = newRow.filter((el) => el !== 0);
+
+        while (newRow.length < 4) {
+          newRow.push(0);
+        }
+
+        return newRow.reverse();
+      });
+
+      if (this.checkWin()) {
+        this.status = 'win';
+      }
+
+      if (this.differenceArrays(currentState, this.getState())) {
+        this.checkRight = true;
+      } else {
+        this.checkRight = false;
+      }
+
+      if (this.checkRight) {
+        this.addRandomTile(this.board);
+      }
+
+      if (this.checkLose() && !this.checkWin()) {
+        this.status = 'lose';
+      }
```

Comment:
you can move this logic 
<img width="670" alt="image" src="https://github.com/user-attachments/assets/5a3a2c81-3dbc-4836-9166-2d1a41636a7c">
and simplify these 4 functions